src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1531,12 +1531,14 @@ NewTableId(Index originalTableId, List *rangeTableList)
 		if (listMember)
 		{
 			newTableId = rangeTableIndex;
-			break;
+			return newTableId;
 		}
 
 		rangeTableIndex++;
 	}
 
+	ereport(ERROR, (errmsg("Unrecognized range table id %d", (int) originalTableId)));
+
 	return newTableId;
```

Comment:
Wouldn't it be better to remove the variable and return directly from the loop? Because this return is dead code...